src/netex/read.rs

```diff
@@ -15,9 +15,194 @@
 // <http://www.gnu.org/licenses/>.
 
 use model::Collections;
+use objects;
 use std::io::Read;
 use Result;
 
-pub fn read_netex_file<R: Read>(_collections: &mut Collections, _file: R) -> Result<()> {
-    Ok(())
+extern crate minidom;
+extern crate serde_json;
+use self::minidom::Element;
+
+// type RoutePointId = String;
+// type StopPointId = String;
+// type RoutePointMapping = HashMap<RoutePointId, StopPointId>;
+// type RouteLineMap = HashMap<String, String>;
+
+#[derive(Default)]
+struct NetexContext {
+    namespace: String,
+    first_operator_id: String,
+    // network_id: String,
+    // routepoint_mapping: RoutePointMapping,
+    // route_line_map: RouteLineMap,
+    // route_mode_map: HashMap<String, String>,
+    // journeypattern_route_map: HashMap<String, String>,
+}
+
+#[derive(Default)]
+pub struct NetexReader {
+    context: NetexContext,
+    pub collections: Collections,
+}
+impl NetexReader {
+    pub fn read_netex_file<R: Read>(&mut self, mut file: R) -> Result<()> {
+        let mut file_content = "".to_string();
+        file.read_to_string(&mut file_content)?;
+        let root: Element = file_content.parse()?;
+
+        self.context.namespace = root.ns().unwrap_or("".to_string());
+
+        root.get_child("dataObjects", self.context.namespace.as_str())
+            .map_or_else(
+                || bail!("Netex file does't contain a 'dataObjects' node"),
+                |f| {
+                    f.children()
+                        .filter(|frame| frame.name() == "CompositeFrame")
+                        .map(|frame| {
+                            self.read_composite_data_frame(frame).map_err(|_| {
+                                format_err!(
+                                    "Reading Frame id={:?}",
+                                    frame.attr("id").unwrap_or("undefined")
+                                )
+                            })
+                        })
+                        .collect()
+                },
+            )
```

Comment:
Wouhaou! I take some time to understand that...

Better to do that simpler:

```rust
for frame in root
    .get_child("dataObjects", self.context.namespace.as_str())
    .ok_or_else(|| bail!("Netex file does't contain a 'dataObjects' node"))?
    .filter(|frame| frame.name() == "CompositeFrame")
{
    self.read_composite_data_frame(frame)
        .with_context(|_| format!("Reading Frame id={:?}", frame.attr("id").unwrap_or("undefined")))?
}
Ok(())
```

Reply:
@TeXitoi I take some time to understand your code and I still don't understand

Reply:
@datanel you prefer the original version? you have a better idea?

Reply:
I understand now :wink: 
@prhod don't forget `use failure::ResultExt;`